[SKarthikeyan28] iP

SKarthikeyan28 opened the pull request:
# TARS

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))
Welcome to TARS, you're one stop task manager with many exciting features! 

Why TARS?
* text-based
* easy to learn
* ~FAST~ SUPER FAST to use

All you need to do is,

1. download the *.jar* file from [here](https://github.com/SKarthikeyan28/ip/releases/tag/A-Jar).
2. Open the jar file in your IDE
3. Add your tasks
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [ ] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Managing Event reminders (coming soon)

 If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

SKarthikeyan28 committed:
Changed Duke to Tars and welcome message

SKarthikeyan28 committed:
Added commands to echo user

SKarthikeyan28 committed:
Added list variable to store inputs and display

SKarthikeyan28 committed:
Added mark and unmark methods using TASK class

SKarthikeyan28 committed:
Added Event, Deadline and ToDos classes and added to main

SKarthikeyan28 committed:
Conducted Text Testing

SKarthikeyan28 committed:
added exceptions

SKarthikeyan28 committed:
Added delete option

SKarthikeyan28 committed:
Added javadoc comments

SKarthikeyan28 committed:
Added storage feature for Tars application

SKarthikeyan28 committed:
Txt file to save data of tasks in list

SKarthikeyan28 committed:
Added Date and Time handling method

SKarthikeyan28 committed:
Refactor classes to utilise more OOP

SKarthikeyan28 committed:
Add package Tars to all classes

SKarthikeyan28 committed:
Merge commit '81a9c537afe8492675eb253968939364d65db091'

* commit '81a9c537afe8492675eb253968939364d65db091':
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

SKarthikeyan28 committed:
Add Gradle

SKarthikeyan28 committed:
Add Test Class for Tars

SKarthikeyan28 committed:
Add shadowJar to build.gradle

SKarthikeyan28 committed:
Add JavaDoc comments for Tars classes

SKarthikeyan28 committed:
Edit code to match Coding Standard

SKarthikeyan28 committed:
Add find feature to Tars

SKarthikeyan28 committed:
Merge branch 'branch-A-CodingStandard'

SKarthikeyan28 committed:
Merge branch 'branch-Level-9'

SKarthikeyan28 committed:
Add Checkstyle to project and check code style

SKarthikeyan28 committed:
Add GUI functionality to Tars

SKarthikeyan28 committed:
Merge branch 'branch-Level-10' to master

meatyturtle commented on README.md:
```diff
@@ -1,4 +1,4 @@
-# Duke project template
+# Tars project template
```
looks good! the PR description is a nice personal touch

meatyturtle commented on data/Tars.txt:
```diff
@@ -0,0 +1,13 @@
+T 0 return book
+T 0 buy book
+E 0 hi (from: 2019-10-15T14:30 to: 2019-10-16T15:30 )
```
not a coding standard issue, but you could consider formatting the dates and times in a more friendly readable format

meatyturtle commented on src/main/java/tars/Parser.java:
```diff
@@ -0,0 +1,95 @@
+     * Chooses appropriate methods to call based on input given or throws error message
+     * depending on incorrect input received
+     *
+     * @param entryParts
```
you could write a short phrase to describe the params and return value for clarity
e.g. @param entry The input string from the user
@param tasks The list of tasks

meatyturtle commented on src/main/java/tars/Parser.java:
```diff
@@ -0,0 +1,95 @@
+
+    /**
+     * Finds Task based on keyword stated by User, takes in entry of users and tasks in list
+     * Goes through all the tasks in the list that match keyword stated by User
```
you could be more specific with the header comment, i.e. instead of using "goes through all the tasks" you could mention that it returns the task that keywords match
i know that you mean iterate through the list of tasks but it would be more concise to just describe what the method is supposed to do :D

meatyturtle commented on src/main/java/tars/Parser.java:
```diff
@@ -0,0 +1,95 @@
+        }
+
+        for (int i = 0; i < resultList.size(); i++) {
+            strBuild.append("    " + (i + 1) + ". " + resultList.get(i) + "\n");
```
if you're using IntelliJ IDEA, i think it would suggest to use chained append commands rather than '+', but either way it is readable and easy to understand.

meatyturtle commented on src/main/java/tars/Storage.java:
```diff
@@ -0,0 +1,116 @@
+    }
+
+    /**
+     * Writes file that will be used for the Tars application based on tasks existing or changed in tasksList
```
you could be more concise with this header comment, e.g. something like
Writes existing tasks or changes in the tasklist to file

meatyturtle commented on src/main/java/tars/Storage.java:
```diff
@@ -0,0 +1,116 @@
+     * @throws IOException
+     */
+    public static ArrayList<Task> readFile() throws IOException {
+        Scanner s = new Scanner(taskFile); // create a Scanner using the File as the source
```
since this is not temporary, you may want to give a more descriptive name like scanner (as compared to s) for clarity
not crucial since scanner is not used outside of this file 😛

meatyturtle commented on src/main/java/tars/Tars.java:
```diff
@@ -0,0 +1,99 @@
+     */
+    public Tars(String filePath) throws IOException {
+        this.ui = new Ui();
+        storage = new Storage(filePath);
```
is Storage static? else you could be consistent and use the `this` keyword, as with UI and Parser. (correct me if im wrong i might have missed it)

meatyturtle commented on src/main/java/tars/Tars.java:
```diff
@@ -0,0 +1,99 @@
+     *
+     * @throws IOException
+     */
+    /*public void run() throws IOException {
```
do remove this block comment if you decide to scrap this method to ensure your code looks clean (it will be here, on github if you want to recover it)

meatyturtle commented on src/main/java/tars/Tars.java:
```diff
@@ -0,0 +1,99 @@
+        return result;
+    }
+
+    /**
```
same with this, remove if you dont want it anymore 👍

meatyturtle commented on src/main/java/tars/Task.java:
```diff
@@ -0,0 +1,196 @@
+     *
+     * @return "X" if task is done, " " otherwise returned
+     */
+    public String getStatusIcon() {
```
since you used the `this` keyword to refer to the instance of isDone in toher places you should use it here too, for consistency

meatyturtle commented on src/main/java/tars/Task.java:
```diff
@@ -0,0 +1,196 @@
+*
+* @see Task
+*/
+class ToDos extends Task {
```
this may be a major change so i understand if you wouldnt want to do it xD, but consider renaming this to ToDo (singular) instead. reason is every time you instantiate this class only one instance of a Todo task is created, as opposed to a Collection of ToDo (like a list or something)

meatyturtle commented on src/main/java/tars/Task.java:
```diff
@@ -0,0 +1,196 @@
+ *
+ * @see Task
+ */
+class Deadline extends Task {
```
consider putting each type of task in its own file, then create a package for tasks

meatyturtle reviewed (commented):
looks good! the pr description is a nice personal touch, just some minor coding standard issues to help with clarity

DennieDan commented on src/main/java/tars/MainWindow.java:
```diff
@@ -0,0 +1,55 @@
+    private Tars tars;
+
+    private Image userImage = new Image(this.getClass().getResourceAsStream("/images/DaUser.png"));
+    private Image dukeImage = new Image(this.getClass().getResourceAsStream("/images/DaDuke.png"));
```
Congrats on successfully adding GUI to your project! Now it's time for you to change those names from **duke** into your **tars**, isn't it?

DennieDan commented on src/main/java/tars/Parser.java:
```diff
@@ -0,0 +1,95 @@
+     *
+     * @param tasks
+     */
+    public String listPrint(TaskList tasks) {
```
I think this method's name is supposed to be ``printList`` instead, right? i.e. in the verb form 😺

DennieDan commented on src/main/java/tars/Tars.java:
```diff
@@ -0,0 +1,99 @@
+    private TaskList taskList;
+    private Storage storage;
+    private Ui ui;
+    private Parser parserHelp;
```
Do you have any other parsers in this file? If not, I think call this ``parser`` is easier to understand ah!

DennieDan commented on src/main/java/tars/TaskList.java:
```diff
@@ -0,0 +1,199 @@
+
+            result = LINE + "    OK, I've marked this task as not done yet:" + taskList.get(index - 1) + "\n" + LINE;
+        } else if (task[0].equals("todo")) {
+            result = this.addToDos(task, entry);
```
I am wondering why ``addToDos`` ends with an **s** while ``addDeadline`` and ``addEvent`` don't?

DennieDan reviewed (commented):
Most of the names look great!
Some very minors mistakes to improve your Code Quality: Naming

SKarthikeyan28 committed:
Assert Feature: Add Assert Feature to different classes to use for
debugging.

Assert statements help with identifying bugs related to assumptions
made about certain values or actions. This will be useful when
invalid input entries are given by users which could result in the
code not running properly.

Let's check arguments that are using user inputs and task list
as arguments to verify if any invalid inputs or empty lists are
passed as values.

Checking these values will be essential as these are the common
values and arguments used in the code to execute other methods
and commands for user to receive output.

SKarthikeyan28 committed:
Edit code to match code quality requirements

Some of the code in some classes were long, this makes the code
hard to read. Some of the code were also using complicated
expressions which might be hard to understand too.

Let's simplify them by refactoring repeated logic into a separate
method and use it instead of repeating the lines at multiple
instances. Lets also simplify the code blocks by breaking the
boolean conditions into smaller blocks.

This will greatly help in increasing readability of code which is
essential for readers to understand at one glance and when also
revisiting code back after long time

SKarthikeyan28 committed:
Merge pull request #1 from SKarthikeyan28/branch-A-Assertions

Merge branch-A-Assertions to master branch

SKarthikeyan28 committed:
Merge branch 'master' into branch-A-CodeQuality

SKarthikeyan28 committed:
Merge pull request #2 from SKarthikeyan28/branch-A-CodeQuality

Merge branch-A-CodeQuality to master

SKarthikeyan28 committed:
Add C-FriendlierSyntax extension to Tars by simplifying input commands

SKarthikeyan28 committed:
Merge branch-BCD-Extension to master

SKarthikeyan28 committed:
Edit Readme for Tars giving comprehensive guide

SKarthikeyan28 committed:
Add colour to background of GUI and resizing ability

SKarthikeyan28 committed:
Refactor getting description and date for tasks to separate methods

SKarthikeyan28 committed:
Add more test cases to junit test classes to test taskList and Parser

SKarthikeyan28 committed:
Edit readMe doc with more information on command instructions

SKarthikeyan28 committed:
Edit findTask method to better handle partial inputs

SKarthikeyan28 committed:
Add Javadoc comments and credit for Chat-GPT assisted code

SKarthikeyan28 committed:
Tweak output messages for some methods to match overall format
